### PR TITLE
Improve storage credential explanation

### PR DESCRIPTION
**Pull request by rosewms**
## Description

* Edit info about Hadoop's `core-site.xml` and improve clarify 

> Is this change a fix, improvement, new feature, refactoring, or other? Improvement

> Is this a change to the core query engine, a connector, client library, or the SPI interfaces? Hive Azure doc

> How would you describe this change to a non-technical end user or system administrator? see doc edit

## Documentation

( ) No documentation is needed.
(x) Sufficient documentation is included in this PR.
( ) Documentation PR is available with #prnumber.
( ) Documentation issue #issuenumber is filed, and can be handled later.

## Release notes

(x) No release notes entries required.
( ) Release notes entries required with the following suggested text:

### CODE REVIEW

File: docs/src/main/sphinx/connector/hive-azure.rst [hunk 1]
```diff
@@ -15,26 +15,32 @@ we recommend using ADLS Gen2. Learn more from `the official documentation
...
+  If you have the ``core-site.xml``, you are not required to independently
+  configure Trino with your storage credentials.
+
 Configuration for Azure Storage credentials
 -------------------------------------------
 
-If you do not want to rely on Hadoop's ``core-site.xml`` and want to have Trino
-configured independently with the storage credentials, you can use the following
-properties in the catalog configuration.
+To configure Trino, use the following configuration properties in the catalog
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/hive-azure.rst:37]: This is the general recommendation .. it should move higher up

File: docs/src/main/sphinx/connector/hive-azure.rst [hunk 2]
```diff
@@ -15,26 +15,32 @@ we recommend using ADLS Gen2. Learn more from `the official documentation
...
 ----------------------------
 
-All configuration for the Azure storage driver is stored in the Hadoop
-``core-site.xml`` configuration file. The path to the file needs to be
-configured in the catalog properties file:
+If you have Hadoop's ``core-site.xml``, we recommend using it. It has all of the
+settings you need to point to the ``site.xml``.
+
+All of the configuration properties for the Azure storage driver are stored in
+the Hadoop ``core-site.xml`` configuration file. The path to the file needs to
```
Comment (m57lyra) [inline docs/src/main/sphinx/connector/hive-azure.rst:22]: must be

File: docs/src/main/sphinx/connector/hive-azure.rst (hunk 1 above)
Comment (m57lyra) [inline docs/src/main/sphinx/connector/hive-azure.rst:37]: To configure Trino to use Azure Storage credentials

File: docs/src/main/sphinx/connector/hive-azure.rst [hunk 3]
```diff
@@ -15,26 +15,32 @@ we recommend using ADLS Gen2. Learn more from `the official documentation
...
+  configure Trino with your storage credentials.
+
 Configuration for Azure Storage credentials
 -------------------------------------------
 
-If you do not want to rely on Hadoop's ``core-site.xml`` and want to have Trino
-configured independently with the storage credentials, you can use the following
-properties in the catalog configuration.
+To configure Trino, use the following configuration properties in the catalog
+properties file. We recommend using this type of configuration when the Primary
```
Comment (m57lyra) [inline docs/src/main/sphinx/connector/hive-azure.rst:38]: primary storage account

(don't productize)

**Review by m57lyra — APPROVED** [review]

File: docs/src/main/sphinx/connector/hive-azure.rst [hunk 4]
```diff
@@ -12,29 +12,27 @@ Trino supports both ADLS Gen1 and Gen2. With ADLS Gen2 now generally available,
...
-``core-site.xml`` configuration file. The path to the file needs to be
-configured in the catalog properties file:
+To configure Trino to use Azure Storage credentials, set the following
+configuration properties in the catalog properties file. It is best to use this
+type of configuration if the primary storage account is linked to the cluster.
 
-.. code-block:: text
+If you are including secondary storage accounts, we recommend configuring Trino
+with the ``core-site.xml``. It contains the credentials for each account, as
+described in the previous section.
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/hive-azure.rst:24]: that makes no sense now

File: docs/src/main/sphinx/connector/hive-azure.rst [hunk 5]
```diff
@@ -12,29 +12,27 @@ Trino supports both ADLS Gen1 and Gen2. With ADLS Gen2 now generally available,
...
 
-    hive.config.resources=<path_to_hadoop_core-site.xml>
+Additionally, all of the configuration properties for the Azure storage driver
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/hive-azure.rst:26]: that paragraph content overlaps with prior paragraph.. collapse them please

File: docs/src/main/sphinx/connector/hive-azure.rst [hunk 6]
```diff
@@ -12,29 +12,27 @@ Trino supports both ADLS Gen1 and Gen2. With ADLS Gen2 now generally available,
...
-----------------------------
+Hive connector configuration for Azure Storage credentials
+----------------------------------------------------------
 
-All configuration for the Azure storage driver is stored in the Hadoop
-``core-site.xml`` configuration file. The path to the file needs to be
-configured in the catalog properties file:
+To configure Trino to use Azure Storage credentials, set the following
+configuration properties in the catalog properties file. It is best to use this
+type of configuration if the primary storage account is linked to the cluster.
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/hive-azure.rst:20]: It might be worth explaining that now core-site.xml file is needed in this case

**Review by mosabua — CHANGES_REQUESTED** [review]

File: docs/src/main/sphinx/connector/hive-azure.rst (hunk 6 above)
Comment (rosewms) [reply docs/src/main/sphinx/connector/hive-azure.rst:20]: Let me know if this restructuring of information is more accurate.

File: docs/src/main/sphinx/connector/hive-azure.rst [hunk 7]
```diff
@@ -12,29 +12,24 @@ Trino supports both ADLS Gen1 and Gen2. With ADLS Gen2 now generally available,
...
-configured independently with the storage credentials, you can use the following
-properties in the catalog configuration.
+.. code-block:: text
 
-We suggest to use this kind of configuration when you only have the Primary
-Storage account linked to the cluster. When there are secondary storage
-accounts involved, we recommend configuring Trino using a ``core-site.xml``
-containing the appropriate credentials for each account, as described in the
-preceding section.
+    hive.config.resources=<path_to_hadoop_core-site.xml>
```
Comment (findinpath) [inline docs/src/main/sphinx/connector/hive-azure.rst:32]: I haven't used so far `core-site.xml` to connect to abfs. I think it would be beneficial to have a very simplistic snippet out of a `core-site.xml` file listed below to give a feeling to a newbie user how this file looks like.

File: docs/src/main/sphinx/connector/hive-azure.rst [hunk 8]
```diff
@@ -12,29 +12,24 @@ Trino supports both ADLS Gen1 and Gen2. With ADLS Gen2 now generally available,
...
+All of the configuration properties for the Azure storage driver are stored in
+the Hadoop ``core-site.xml`` configuration file. To configure Trino to use the
+Azure Storage credentials, set the following configuration properties in the
+catalog properties file. It is best to use this type of configuration if the
+primary storage account is linked to the cluster.
 
-    hive.config.resources=<path_to_hadoop_core-site.xml>
+When there are secondary storage accounts involved, we recommend configuring
+Trino using a ``core-site.xml`` containing the appropriate credentials for each
+account, as described in the previous section.
```
Comment (findinpath) [inline docs/src/main/sphinx/connector/hive-azure.rst:26]: What do you mean in this context by _previous section_ ?
This seems to me the "main" section. 
https://trino.io/docs/current/connector/hive-azure.html

File: docs/src/main/sphinx/connector/hive-azure.rst (hunk 7 above)
Comment (mfzarko) [reply docs/src/main/sphinx/connector/hive-azure.rst:32]: I agree here, a minimal example would be great here.

**Review by mfzarko — CHANGES_REQUESTED** [review]
LGTM, 
I feel a `core-site.xml` config snippet as an example would best.

File: docs/src/main/sphinx/connector/hive-azure.rst (hunk 7 above)
Comment (rosewms) [reply docs/src/main/sphinx/connector/hive-azure.rst:32]: @mfzarko is there a specific example you have in mind that I can copy-paste?

Comment (mfzarko) [reply docs/src/main/sphinx/connector/hive-azure.rst:32]: Sure, 
look here: https://hadoop.apache.org/docs/stable/hadoop-azure/abfs.html

```
The abfswales1.dfs.core.windows.net account is the name by which the storage account will be referred to.

Now ask for the connection string to the store, which contains the account key

az storage account  show-connection-string --name abfswales1
{
  "connectionString": "DefaultEndpointsProtocol=https;EndpointSuffix=core.windows.net;AccountName=abfswales1;AccountKey=ZGlkIHlvdSByZWFsbHkgdGhpbmsgSSB3YXMgZ29pbmcgdG8gcHV0IGEga2V5IGluIGhlcmU/IA=="
}
You then need to add the access key to your core-site.xml, JCEKs file or use your cluster management tool to set it the option fs.azure.account.key.STORAGE-ACCOUNT to this value.

<property>
  <name>fs.azure.account.key.abfswales1.dfs.core.windows.net</name>
  <value>ZGlkIHlvdSByZWFsbHkgdGhpbmsgSSB3YXMgZ29pbmcgdG8gcHV0IGEga2V5IGluIGhlcmU/IA==</value>
</property>

```

File: docs/src/main/sphinx/connector/hive-azure.rst [hunk 9]
```diff
@@ -12,29 +12,51 @@ Trino supports both ADLS Gen1 and Gen2. With ADLS Gen2 now generally available,
...
-configured in the catalog properties file:
+The path to the file must be configured in the catalog properties file:
 
 .. code-block:: text
 
     hive.config.resources=<path_to_hadoop_core-site.xml>
 
-Configuration for Azure Storage credentials
--------------------------------------------
+Example
```
Comment (rosewms) [inline docs/src/main/sphinx/connector/hive-azure.rst:34]: @findinpath and @mfzarko how's this? I created an example section with your suggestion. 

@mosabua from a structure standpoint, do you agree with the example subsection style?

Comment (mfzarko) [reply docs/src/main/sphinx/connector/hive-azure.rst:34]: LGTM

**Review by mfzarko — APPROVED** [review]
LGTM

File: docs/src/main/sphinx/connector/hive-azure.rst [hunk 10]
```diff
@@ -12,29 +12,51 @@ Trino supports both ADLS Gen1 and Gen2. With ADLS Gen2 now generally available,
 we recommend using ADLS Gen2. Learn more from `the official documentation
 <https://docs.microsoft.com/en-us/azure/data-lake-store/data-lake-store-overview>`_.
 
-Hive connector configuration
-----------------------------
+Hive connector configuration for Azure Storage credentials
+----------------------------------------------------------
+
+All of the configuration properties for the Azure storage driver are stored in
+the Hadoop ``core-site.xml`` configuration file. To configure Trino to use the
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/hive-azure.rst:19]: Is that true.. does the connector actually create and update that file?

File: docs/src/main/sphinx/connector/hive-azure.rst [hunk 11]
```diff
@@ -12,29 +12,51 @@ Trino supports both ADLS Gen1 and Gen2. With ADLS Gen2 now generally available,
...
--------------------------------------------
+Example
+^^^^^^^
+
+The ``abfsexample.dfs.core.windows.net`` account refers to the storage account.
+The connection string to the store contains the account key:
+
+.. code-block:: text
+
+    az storage account  show-connection-string --name abfswales1
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/hive-azure.rst:42]: what is this snippet .. please explain ..(uses the az command to do what?)

File: docs/src/main/sphinx/connector/hive-azure.rst [hunk 12]
```diff
@@ -12,29 +12,51 @@ Trino supports both ADLS Gen1 and Gen2. With ADLS Gen2 now generally available,
...
+    {
+      "connectionString": "DefaultEndpointsProtocol=https;EndpointSuffix=core.windows.net;AccountName=abfsexample;AccountKey=examplekey..."
+    }
+
+Add the access key to your ``core-site.xml``, JCEKs file or use your cluster
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/hive-azure.rst:47]: JCEKs .. what is that and why is that in context here

File: docs/src/main/sphinx/connector/hive-azure.rst [hunk 13]
```diff
@@ -12,29 +12,51 @@ Trino supports both ADLS Gen1 and Gen2. With ADLS Gen2 now generally available,
...
+management tool to set it the option ``fs.azure.account.key.STORAGE-ACCOUNT`` to
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/hive-azure.rst:48]: set it the option ??

**Review by mosabua — CHANGES_REQUESTED** [review]
This adds more info at this stage but does not make anything clearer. Please revisit.

File: docs/src/main/sphinx/connector/hive-azure.rst (hunk 10 above)
Comment (rosewms) [reply docs/src/main/sphinx/connector/hive-azure.rst:19]: @mfzarko can you confirm this is the functionality?

Comment (mfzarko) [reply docs/src/main/sphinx/connector/hive-azure.rst:19]: @mosabua @rosewms 
Since there's no JIRA attached it's hard to follow what is the primary goal, I can only attest to technical scope here.

I can  confirm what Manfred said: `catalog.properties` take precedence before `core-site.xml`, which should be consumed only as last resort.
IIRC the plan was to retire `core-site.xml` entirely where possible, but just-in-case reference the values that **need** to be configured through it.
Rose, could you clarify the overall intention here?

**Comment by rosewms** [comment]
@mosabua updated PR with the reorganized content to emphasize using the catalog properties. Additionally clarifies the `core-site.xml`, that information has been moved to the bottom of the "Hive connector configuration for Azure Storage credentials" section.

File: docs/src/main/sphinx/connector/hive-azure.rst [hunk 14]
```diff
@@ -12,30 +12,16 @@ Trino supports both ADLS Gen1 and Gen2. With ADLS Gen2 now generally available,
...
-accounts involved, we recommend configuring Trino using a ``core-site.xml``
-containing the appropriate credentials for each account, as described in the
-preceding section.
+To configure Trino to use the Azure Storage credentials, set the following
+configuration properties in the catalog properties file. It is best to use this
+type of configuration if the primary storage account is linked to the cluster.
 
+The specific configuration depends on the type of storage and uses the
+properties from the following sections in the catalog properties file.
+
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/hive-azure.rst:24]: just add a sentence here that mentions that in more complex cases you can use the core-site xml (and link to that section)..

**Review by mosabua — CHANGES_REQUESTED** [review]

**Review by electrum — APPROVED** [review]
